adal/src/main/java/com/microsoft/aad/adal/PRNGFixes.java

```diff
@@ -244,14 +249,22 @@ protected void engineNextBytes(byte[] bytes) {
                 // Mix in the device- and invocation-specific seed.
                 engineSetSeed(generateSeed());
             }
+            DataInputStream in = null;
             try {
-                DataInputStream in;
+                in = getUrandomInputStream();
                 synchronized (SLOCK) {
-                    in = getUrandomInputStream();
                     in.readFully(bytes);
                 }
             } catch (IOException e) {
                 throw new SecurityException("Failed to read from " + URANDOM_FILE, e);
+            } finally {
+                if (in != null) {
+                    try {
+                        in.close();
```

Comment:
Strange this got missed. Thanks for making this fix 👍

Reply:
thanks for the 'ship it' :)